dvc/output/base.py

```diff
@@ -282,6 +282,9 @@ def save(self):
         self.hash_info = self.get_hash()
 
     def commit(self):
+        if not self.exists:
+            raise self.DoesNotExistError(self)
```

Comment:
Unsure if it would be better for us to skip this exists call for performance reasons, and just catch (and allow) any exception on `commit()` in `Stage.run()` when we want `allow_exists` behavior?

Reply:
Yeah, but the problem is that tree exceptions are not unified yet, so it will be all over the place :slightly_frowning_face: But since you are only working with local trees, you might skip this check and just catch FileNotFoundError even right now. Up to you.